.github/workflows/build_workflow.yml

```diff
@@ -0,0 +1,262 @@
+
+      - name: Setup Java 21 LTS
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'temurin'
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+
+      - name: Spotbugs
+        run: |
+          ./gradlew spotbugsMain spotbugsTest --parallel
+        working-directory: ./service
+
+      - name: Collect Artifacts
+        if: always()
+        run: |
+          mkdir -p artifacts
+          cp -r ./service/build/reports ./artifacts
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: 'Spotbugs Reports'
+          path: ./artifacts/**
+          compression-level: 9
+
+      - name: Temporary Artifacts Cleanup
+        run: rm -rf ./artifacts
+
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    needs: [ checkstyle, spotbugs ]
+    steps:
+      -   name: Checkout Repository
+          uses: actions/checkout@v4
+          with:
+            # Disabling shallow clone is recommended for improving relevancy of Sonar reporting
+            fetch-depth: 0
+      -   name: Setup Java 21 LTS
+          uses: actions/setup-java@v4
+          with:
+            java-version: 21
+            distribution: 'temurin'
+
+      -   name: Setup Gradle
+          uses: gradle/actions/setup-gradle@v4
+
+      -   name: Execute Unit Tests
+          run: ./gradlew test jacocoTestReport --parallel --build-cache
+          working-directory: ./service
+
+      -   name: Collect Artifacts
+          if: always()
+          run: |
+            mkdir -p artifacts
+            cp -r ./service/build/reports ./artifacts
+
+      -   name: Upload Artifacts
+          uses: actions/upload-artifact@v4
+          if: always()
+          with:
+            name: 'Unit Test Reports'
+            path: ./artifacts/**
+            compression-level: 9
+
+      -   name: Temporary Artifacts Cleanup
+          run: rm -rf ./artifacts
+
+  integration-tests:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    needs: [ checkstyle, spotbugs ]
+    steps:
+      -   name: Checkout Repository
+          uses: actions/checkout@v4
+
+      -   name: Setup Java 21 LTS
+          uses: actions/setup-java@v4
+          with:
+            java-version: 21
+            distribution: 'temurin'
+
+      -   name: Setup Gradle
+          uses: gradle/actions/setup-gradle@v4
+
+      -   name: Start Docker Dependencies
+          env:
+            GPC_CONSUMER_SERVER_PORT: "8090"
+            GPC_CONSUMER_SDS_URL: "http://wiremock:8080/spine-directory/"
+            GPC_CONSUMER_SDS_APIKEY: "anykey"
+            GPC_CONSUMER_LOGGING_LEVEL: DEBUG
+          run: |
+            docker network create commonforgpc
+            docker compose build
+            docker compose up gpc-consumer gpcc-mocks gpconnect-db gpconnect-api tkw gpcc-nginx --detach
+          working-directory: ./docker
```

Comment:
Out of curiosity, did you see why we are using the demonstrator / tkw / gpconnect-db things?

Reply:
Here is a definition of TKW simulator:
The TKW (ToolKit Workbench) Simulators are tools to assist with developing and assuring a solution to meet the Booking Standard. 

As far as I can see these are "dead" projects. they can be found in nhsdigitalmait (https://github.com/nhsdigitalmait) organization. I don't think we need them as the projects do not seem to be developed/maintained anymore. I can clean them up in a separate PR.